source/slang/slang-c-like-source-emitter.cpp

```diff
@@ -0,0 +1,5811 @@
+// slang-c-like-source-emitter.cpp
+#include "slang-c-like-source-emitter.h"
```

Comment:
Naming nit: I might vote for slightly more compact naming:

```
emit-c-like.cpp
emit-hlsl.cpp
emit-glsl.cpp
...
```

That organization puts the common part (`emit`) toward the start so that files sort nicely (similar to how `ir-` gets used as a prefix for (almost) everything IR-related). I also like dropping extra parts from the file name if possible (I'd argue `slang` isn't adding anything here since we don't prefix our other files with it; `source` is implied by `c-like` since all c-like languages are source).